### PR TITLE
Dump SeaORM Dependency from SeaSchema's Migrator

### PR DESCRIPTION
**Pull request by billy1624**
## PR Info

- Closes SeaQL/sea-schema#57
- Dependencies:
  - SeaQL/sea-schema#59

## Adds

- All migration traits and structs are moved to SeaOMR except SchemaManager

## Breaking Changes

- `sea_schema::migration::prelude` should be replaced by `sea_orm::migration::prelude`

### CODE REVIEW

**Comment by tyt2y3** [comment]
This changeset is too big and it kind of broke the API for no good reason. Is there another approach?

**Comment by billy1624** [comment]
What you mean by "breaking the API"?

**Comment by billy1624** [comment]
Hey @tyt2y3, this is ready for review!

File: examples/graphql_example/migration/src/lib.rs [hunk 1]
```diff
@@ -1,4 +1,4 @@
-pub use sea_schema::migration::*;
+pub use entity::sea_orm::migration::prelude::*;
```
Comment (tyt2y3) [inline examples/graphql_example/migration/src/lib.rs:1]: why is this `entity::sea_orm` instead of `sea_orm`?

File: Cargo.toml [hunk 2]
```diff
@@ -57,6 +62,7 @@ pretty_assertions = { version = "^0.7" }
 debug-print = []
 default = [
     "macros",
+    "migration",
```
Comment (tyt2y3) [inline Cargo.toml:65]: I don't think we should add this as default feature

File: Cargo.toml [hunk 3]
```diff
@@ -65,6 +71,7 @@ default = [
 ]
 macros = ["sea-orm-macros"]
 mock = []
+migration = ["sea-schema", "macros", "clap", "tracing-subscriber", "dotenv", "async-std"]
```
Comment (tyt2y3) [inline Cargo.toml:74]: It seems that what we need is one more crate sea-orm-migration

File: Cargo.toml [hunk 4]
```diff
@@ -40,11 +40,16 @@ uuid = { version = "0.8", features = ["serde", "v4"], optional = true }
 ouroboros = "0.15"
 url = "^2.2"
 once_cell = "1.8"
+sea-schema = { git = "https://github.com/SeaQL/sea-schema", branch = "dump-sea-orm-dep", features = [ "migration" ], default-features = false, optional = true }
+async-std = { version = "^1", features = [ "attributes", "tokio1" ], optional = true }
+tracing-subscriber = { version = "0.3", features = ["env-filter"], optional = true }
+dotenv = { version = "^0.15", optional = true }
+clap = { version = "^2.33", optional = true }
```
Comment (tyt2y3) [inline Cargo.toml:47]: These dependencies definitely do not belong here :(

**Review by tyt2y3 — CHANGES_REQUESTED** [review]

File: examples/graphql_example/migration/src/lib.rs (hunk 1 above)
Comment (tyt2y3) [reply examples/graphql_example/migration/src/lib.rs:1]: we should make it `sea_orm_cli::migration` or `sea_orm_migration`

Comment (tyt2y3) [reply examples/graphql_example/migration/src/lib.rs:1]: `entity` is a pure logic crate it does not make sense for `migration` be imported via `entity`

File: Cargo.toml (hunk 2 above)
Comment (tyt2y3) [reply Cargo.toml:65]: actually we should remove this from the main `sea-orm` crate altogether

**Comment by tyt2y3** [comment]
Architecturally, the dependency chain should be `sea-query` (AST) -> `sea-schema` (logic) -> `sea-orm-xxx` (API) -> `sea-orm-cli` (executable)

**Comment by tyt2y3** [comment]
Let's take the time to make it right once and for all

**Comment by tyt2y3** [comment]
> I would refer splitting it into `sea-orm-migration`, as I don't want `sea-orm-cli` depends on `sea-orm`. It'll makes things complicated.

Agree

**Review by tyt2y3 — APPROVED** [review]

**Comment by tyt2y3** [comment]
Hell yeah

**Pull request merged**